lib/util.js

```diff
@@ -751,3 +751,23 @@ exports._errnoException = function(err, syscall, original) {
   e.syscall = syscall;
   return e;
 };
+
+
+exports._detailedException = function(err, syscall, address, port, additional) {
```

Comment:
I think this name is too generic, `createDetailedBindException`, or `exceptionWithHostPort` would be better. Other libraries, like fs, might also need detailed exceptions, but with different properties.